### PR TITLE
common/util: Add utility macros

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
The intent is to replace RxM macro that check local MR bit set and likely/unlikely macro.
The PR is created to cover the comments of PR #3199

Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

**Comment by gladkovdmitry17** [comment]
@a-ilango @shefty Please, check the PR. The PR covers one of your comments in PR #3199 
Thanks in advance!

**Comment by gladkovdmitry17** [comment]
@shefty I'm not sure that it is right place to publish OFI_LIKELY/OFI_UNLIKELY macros. What are your thoughts on that?
@jswaro The changes contains replacement of GNI own likely/unlikely macros to common OFI_LIKELY/OFI_UNLIKELY. Can I ask you to take a look into my changes? Thanks in advance!

File: include/rdma/fabric.h [hunk 1]
```diff
@@ -85,6 +85,15 @@ enum {
 #define FI_VERSION_LT(v1, v2)	((FI_MAJOR(v1) < FI_MAJOR(v2)) || \
 				 (FI_MAJOR(v1) == FI_MAJOR(v2) && FI_MINOR(v1) < FI_MINOR(v2)))
 
+#define OFI_OPS(_fid)		(*_fid)->ops
```
Comment (jswaro) [inline include/rdma/fabric.h]: These macros do no appear to be used anywhere.

**Comment by gladkovdmitry17** [comment]
@jswaro There are usage of OFI_*OPS macros in #3199. The intent of this PR is to cover comments that are received in mentioned PR.
If you prefer to not include these changes into one PR, I can split it. But seems these are too small changes to be splitted into the different PRs.

**Comment by jswaro** [comment]
> There are usage of OFI_*OPS macros in #3199. The intent of this PR is to cover comments that are received in mentioned PR.

Does that mean that the OPS macros have been moved from PR 3199 to this PR? If so, be careful that the PRs are merged in the correct order. 

> If you prefer to not include these changes into one PR, I can split it. But seems these are too small changes to be splitted into the different PRs.

It's fine. I didn't notice that there were 5 distinct commits.

**Comment by gladkovdmitry17** [comment]
> Does that mean that the OPS macros have been moved from PR 3199 to this PR?

Yes, you're right. OPS, OFI_LIKELY and MR_LOCAL macros have been moved from PR #3199.

> If so, be careful that the PRs are merged in the correct order.

Sure. There are a couple of major changes should be made in the 3199 PR before it can be merged. I think the PR 3199 will be merged in a couple of week, not earlier :)

File: include/rdma/fabric.h [hunk 2]
```diff
@@ -85,6 +85,15 @@ enum {
...
+#define OFI_FID_OPS(_fid)	(*_fid)->fid.ops
+#define OFI_MR_OPS(_fid)	(*_fid)->mr
+#define OFI_CM_OPS(_fid)	(*_fid)->cm
+#define OFI_MSG_OPS(_fid)	(*_fid)->msg
+#define OFI_RMA_OPS(_fid)	(*_fid)->rma
+#define OFI_TAG_OPS(_fid)	(*_fid)->tagged
+#define OFI_ATOM_OPS(_fid)	(*_fid)->atomic
```
Comment (a-ilango) [inline include/rdma/fabric.h]: These macros doesn't seem to be very useful IMO.

Comment (jswaro) [inline include/rdma/fabric.h]: What is the motivator for the macros anyway?

Comment (shefty) [inline include/rdma/fabric.h]: I'm fine adding macros that will get used by another PR.  But I agree that these don't seem very useful and appear to increase the amount of typing needed.

Comment (gladkovdmitry17) [inline include/rdma/fabric.h]: Ok, seems, it is really won't get used by providers :) And as @shefty noted it requires the more amount of typing needed. Let's remove useless macros.

File: include/rdma/fabric.h (hunk 1 above)
Comment (gladkovdmitry17) [inline include/rdma/fabric.h]: will be removed

**Comment by gladkovdmitry17** [comment]
> but do we want the utility providers to just require API 1.5?

Sure, nice finding. Seems, the validation of `api_version` in `ofi_check_fabric_attr` is enough to avoid further checking.

**Comment by a-ilango** [comment]
>I'm fine with this commit (34de406), but do we want the utility providers to just require API 1.5? Currently, all providers in the upstream master support 1.5.
Updated: added commit number, since I'm not sure if it's obvious based on the review submission.

Commit a1b83820f008603c491c2cafc608eecf8672dadd which is updated version of 34de406 breaks MPICH-CH3 over RxM provider.

Isn't `info->fabric_attr->api_version` same as the version passed by app in fi_getinfo? If the app has requested 1.4 API, we would have to check for `info->mode & FI_LOCAL_MR`.

**Comment by shefty** [comment]
fabric_attr->api_version is set by the fi_getinfo() call prior to returning to the caller.  Where the conversion from 1.4 to 1.5 is done doesn't really matter to me.  I would go with whatever is easiest.

**Comment by a-ilango** [comment]
It's easier to have the check in macro itself. I'll prepare a PR that reverts a1b8382 to 34de406.

**Pull request merged**